### ISSUE TITLE
error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]

### ISSUE THREAD

**Issue by jasonMisaq** [andrewrk/libsoundio#147]
```

[john@localhost build]$ make
[  1%] Building C object CMakeFiles/libsoundio_shared.dir/src/jack.c.o
/home/john/src/libsoundio/src/jack.c: In function ‘soundio_jack_init’:
/home/john/src/libsoundio/src/jack.c:920:59: error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     if ((err = jack_set_port_rename_callback(sij->client, port_rename_calllback, si))) {
                                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/john/src/libsoundio/src/jack.h:19:0,
                 from /home/john/src/libsoundio/src/jack.c:8:
/usr/include/jack/jack.h:523:5: note: expected ‘JackPortRenameCallback {aka int (*)(unsigned int,  const char *, const char *, void *)}’ but argument is of type ‘void (*)(jack_port_id_t,  const char *, const char *, void *) {aka void (*)(unsigned int,  const char *, const char *, void *)}’
 int jack_set_port_rename_callback (jack_client_t *client,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
CMakeFiles/libsoundio_shared.dir/build.make:206: recipe for target 'CMakeFiles/libsoundio_shared.dir/src/jack.c.o' failed
make[2]: *** [CMakeFiles/libsoundio_shared.dir/src/jack.c.o] Error 1
CMakeFiles/Makefile2:173: recipe for target 'CMakeFiles/libsoundio_shared.dir/all' failed
make[1]: *** [CMakeFiles/libsoundio_shared.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

I am trying to build libsoundio on Fedora 25 x86_64,  but I am always getting this error. As there is a line that says `cc1:all warnings being treated as errors`, I guess this might be dealt with if I can tell the build program to ignore warnings (rather than treating them as errors and stopping the build). I tried passing the `-w` to make, but I am still getting the errors and cannot build. What should I do?
To get rid of the error I used --ignore-errors option for make, but still it seems to be partial success; see output below: 
```
[john@localhost build]$ make --ignore-errors 
[  1%] Building C object CMakeFiles/libsoundio_shared.dir/src/jack.c.o
/home/john/src/libsoundio/src/jack.c: In function ‘soundio_jack_init’:
/home/john/src/libsoundio/src/jack.c:920:59: error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     if ((err = jack_set_port_rename_callback(sij->client, port_rename_calllback, si))) {
                                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/john/src/libsoundio/src/jack.h:19:0,
                 from /home/john/src/libsoundio/src/jack.c:8:
/usr/include/jack/jack.h:523:5: note: expected ‘JackPortRenameCallback {aka int (*)(unsigned int,  const char *, const char *, void *)}’ but argument is of type ‘void (*)(jack_port_id_t,  const char *, const char *, void *) {aka void (*)(unsigned int,  const char *, const char *, void *)}’
 int jack_set_port_rename_callback (jack_client_t *client,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[  1%] Building C object CMakeFiles/libsoundio_shared.dir/src/pulseaudio.c.o
[  1%] Building C object CMakeFiles/libsoundio_shared.dir/src/alsa.c.o
[  1%] Linking C shared library libsoundio.so
cc: error: CMakeFiles/libsoundio_shared.dir/src/jack.c.o: No such file or directory
[  1%] Built target libsoundio_shared
Scanning dependencies of target sio_list_devices
[  1%] Building C object CMakeFiles/sio_list_devices.dir/example/sio_list_devices.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'sio_list_devices'.  Stop.
[  1%] Built target sio_list_devices
Scanning dependencies of target sio_sine
[  1%] Building C object CMakeFiles/sio_sine.dir/example/sio_sine.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'sio_sine'.  Stop.
[  1%] Built target sio_sine
Scanning dependencies of target libsoundio_static
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/soundio.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/util.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/os.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/dummy.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/channel_layout.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/ring_buffer.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/jack.c.o
/home/john/src/libsoundio/src/jack.c: In function ‘soundio_jack_init’:
/home/john/src/libsoundio/src/jack.c:920:59: error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     if ((err = jack_set_port_rename_callback(sij->client, port_rename_calllback, si))) {
                                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/john/src/libsoundio/src/jack.h:19:0,
                 from /home/john/src/libsoundio/src/jack.c:8:
/usr/include/jack/jack.h:523:5: note: expected ‘JackPortRenameCallback {aka int (*)(unsigned int,  const char *, const char *, void *)}’ but argument is of type ‘void (*)(jack_port_id_t,  const char *, const char *, void *) {aka void (*)(unsigned int,  const char *, const char *, void *)}’
 int jack_set_port_rename_callback (jack_client_t *client,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/pulseaudio.c.o
[  1%] Building C object CMakeFiles/libsoundio_static.dir/src/alsa.c.o
[  1%] Linking C static library libsoundio.a
/usr/bin/ar: CMakeFiles/libsoundio_static.dir/src/jack.c.o: No such file or directory
[  1%] Built target libsoundio_static
Scanning dependencies of target underflow
[  1%] Building C object CMakeFiles/underflow.dir/test/underflow.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'underflow'.  Stop.
[  1%] Built target underflow
Scanning dependencies of target sio_microphone
[  1%] Building C object CMakeFiles/sio_microphone.dir/example/sio_microphone.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'sio_microphone'.  Stop.
[  1%] Built target sio_microphone
Scanning dependencies of target sio_record
[  1%] Building C object CMakeFiles/sio_record.dir/example/sio_record.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'sio_record'.  Stop.
[  1%] Built target sio_record
Scanning dependencies of target unit_tests
[  1%] Building C object CMakeFiles/unit_tests.dir/test/unit_tests.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/soundio.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/util.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/os.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/dummy.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/channel_layout.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/ring_buffer.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/jack.c.o
/home/john/src/libsoundio/src/jack.c: In function ‘soundio_jack_init’:
/home/john/src/libsoundio/src/jack.c:920:59: error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     if ((err = jack_set_port_rename_callback(sij->client, port_rename_calllback, si))) {
                                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/john/src/libsoundio/src/jack.h:19:0,
                 from /home/john/src/libsoundio/src/jack.c:8:
/usr/include/jack/jack.h:523:5: note: expected ‘JackPortRenameCallback {aka int (*)(unsigned int,  const char *, const char *, void *)}’ but argument is of type ‘void (*)(jack_port_id_t,  const char *, const char *, void *) {aka void (*)(unsigned int,  const char *, const char *, void *)}’
 int jack_set_port_rename_callback (jack_client_t *client,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[  1%] Building C object CMakeFiles/unit_tests.dir/src/pulseaudio.c.o
[  1%] Building C object CMakeFiles/unit_tests.dir/src/alsa.c.o
[  1%] Linking C executable unit_tests
cc: error: CMakeFiles/unit_tests.dir/src/jack.c.o: No such file or directory
[  1%] Built target unit_tests
Scanning dependencies of target overflow
[  1%] Building C object CMakeFiles/overflow.dir/test/overflow.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'overflow'.  Stop.
[  1%] Built target overflow
Scanning dependencies of target latency
[  1%] Building C object CMakeFiles/latency.dir/test/latency.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/soundio.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/util.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/os.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/dummy.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/channel_layout.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/ring_buffer.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/jack.c.o
/home/john/src/libsoundio/src/jack.c: In function ‘soundio_jack_init’:
/home/john/src/libsoundio/src/jack.c:920:59: error: passing argument 2 of ‘jack_set_port_rename_callback’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     if ((err = jack_set_port_rename_callback(sij->client, port_rename_calllback, si))) {
                                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/john/src/libsoundio/src/jack.h:19:0,
                 from /home/john/src/libsoundio/src/jack.c:8:
/usr/include/jack/jack.h:523:5: note: expected ‘JackPortRenameCallback {aka int (*)(unsigned int,  const char *, const char *, void *)}’ but argument is of type ‘void (*)(jack_port_id_t,  const char *, const char *, void *) {aka void (*)(unsigned int,  const char *, const char *, void *)}’
 int jack_set_port_rename_callback (jack_client_t *client,
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[  1%] Building C object CMakeFiles/latency.dir/src/pulseaudio.c.o
[  1%] Building C object CMakeFiles/latency.dir/src/alsa.c.o
[  1%] Linking C executable latency
cc: error: CMakeFiles/latency.dir/src/jack.c.o: No such file or directory
[  1%] Built target latency
Scanning dependencies of target backend_disconnect_recover
[  1%] Building C object CMakeFiles/backend_disconnect_recover.dir/test/backend_disconnect_recover.c.o
make[2]: *** No rule to make target 'libsoundio.so.1.1.0', needed by 'backend_disconnect_recover'.  Stop.
[  1%] Built target backend_disconnect_recover
```

**Comment by 0joshuaolson1** [comment]
It might be related to #7?

**Comment by jasonMisaq** [comment]
@0joshuaolson1 , yes I am building from master.

**Comment by 0joshuaolson1** [comment]
Does #7 look relevant to your Jack2 version?

**Comment by jasonMisaq** [comment]
@0joshuaolson1 my jack is 1.9 and I was not able to follow @andrewrk's advice for updating to jack2 as its build failed but I was able to modify source as follows and libsoundio built fine then.   
```
static int port_rename_calllback(jack_port_id_t port_id, /*changed void to int */
        const char *old_name, const char *new_name, void *arg)
{
    struct SoundIoPrivate *si = (struct SoundIoPrivate *)arg;
    notify_devices_change(si);
   /*return a value*/
   return 0;

}
```

**Comment by 0joshuaolson1** [comment]
@andrewrk I'm guessing @jasonMisaq hasn't closed this because you offered (#7) to look into jack1 support if filed as an issue?

**Comment by jasonMisaq** [comment]
I guess this is the same error mentioned in #7. The solution I tried was @andrewrk's suggestion for modifying the source. Thanks. Now, I am closing this.
